sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -68,6 +68,8 @@ case class WindowSpecDefinition(
 
   override def children: Seq[Expression] = partitionSpec ++ orderSpec
 
+  override def checkInputDataTypes(): TypeCheckResult = TypeCheckResult.TypeCheckSuccess
```

Comment:
This is fine. We only need to check the data types for those window functions (for now, we only have HiveWindowFunction).  WindowSpecDefinition and WindowExpression are just containers.